doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -399,6 +415,117 @@ int main(void)
 
 <div id="app-obj-c" class="tab-pane fade" markdown="1" >
 
+```objc
+#import "ViewController.h"
+
+@import Kaa;
+
+@interface ViewController () <KaaClientStateDelegate, ConfigurationDelegate, ProfileContainer>
+
+@property (nonatomic, strong) id<KaaClient> kaaClient;
+@property (nonatomic, weak) NSTimer *samplingTimer;
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    NSLog(@"Your first application started");
+
+    // Create a Kaa client and add a listener which displays the Kaa client status
+    // as soon as the Kaa client is started.
+    self.kaaClient = [KaaClientFactory clientWithStateDelegate:self];
+    
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *configurationPath = [[paths objectAtIndex:0] stringByAppendingPathComponent:@"savedconfig.cfg"];
+
+    // Persist configuration in a local storage to avoid downloading it each time the Kaa client is started.
+    [self.kaaClient setConfigurationStorage:[SimpleConfigurationStorage storageWithPath:configurationPath]];
+    
+    // Add a listener which displays the Kaa client configuration each time it is updated.
+    [self.kaaClient addConfigurationDelegate:self];
+    
+    [self.kaaClient setProfileContainer:self];
+    
+    // Start the Kaa client and connect it to the Kaa server.
+    [self.kaaClient start];
+}
+
+#pragma mark - Supporting methods
+
+- (void)createSamplingTimerWithInterval:(NSInteger)period {
+    if (self.samplingTimer) {
+        [self.samplingTimer invalidate];
```

Comment:
Add next line below (it's a good practice to avoid exceptions calling the instance several times).
```
self.samplingTimer = nil;
```